Relation-SPA/src/app/relation-table/relation-table.component.html

```diff
@@ -2,20 +2,106 @@
-              (click)="sendCategoryId(cat.id)">{{ cat.name }}</a>
+            <a class="dropdown-item" role="button" (click)="loadAllRelations()"
+              >Load All</a
+            >
+            <a
+              class="dropdown-item"
+              role="button"
+              *ngFor="let cat of categories"
+              (click)="sendCategoryId(cat.id)"
+              >{{ cat.name }}</a
+            >
           </li>
         </ul>
       </div>
     </div>
     <div class="col-md-1">
-      <button class="btn btn-success">Add+</button>
+      <button
+        type="button"
+        class="btn btn-success"
+        (click)="openModal(template)"
+      >
+        Add +
+      </button>
+      <ng-template #template>
+        <div class="modal-header">
+          <h4 class="modal-title pull-left">New Relation</h4>
+          <button
+            type="button"
+            class="close pull-right"
+            aria-label="Close"
+            (click)="modalRef.hide()"
+          >
+            <span aria-hidden="true">&times;</span>
+          </button>
+        </div>
+        <div class="modal-body">
+          <form>
+            <div class="form-group">
+              <label for="name">Select Category</label>
+              <select class="form-control" id="name">
+                <option *ngFor="let category of categories">{{category.name}}</option>
+              </select>
+            </div>
+            <div class="form-group">
+              <label for="name">Name</label>
+              <input type="text" id="name" class="form-control" />
+            </div>
+            <div class="form-group">
+              <label for="fullName">FullName</label>
+              <input type="text" id="fullName" class="form-control" />
+            </div>
+            <div class="form-group">
+              <label for="emailAddress">Email</label>
+              <input type="email" id="emailAddress" class="form-control" />
+            </div>
+            <div class="form-group">
+              <label for="telephoneNumber">Telephone Number</label>
+              <input type="text" id="telephoneNumber" class="form-control" />
+            </div>
+            <div class="form-group">
+              <label for="countryName">Select Country</label>
+              <select class="form-control" id="countryName">
+                <option>1</option>
+              </select>
+            </div>
+            <div class="form-group">
+              <label for="city">City</label>
+              <input type="text" id="city" class="form-control" />
+            </div>
+            <div class="form-group">
+              <label for="street">Street</label>
+              <input type="text" id="street" class="form-control" />
+            </div>
+            <div class="form-group">
+              <label for="number">Street Number</label>
+              <input type="text" id="number" class="form-control" />
```

Comment:
Is it stored in db as int? If so we can use type="number".

Reply:
Corrected and now its properly typed as number;